### DISCUSSION TITLE
Debugging through the `dotnet` CLI tool

### DISCUSSION THREAD

**Discussion by rcdailey** [aws/aws-lambda-dotnet#1836] [Q&A] [answered]
I don't really want paths to global tools in my `launchSettings.json`; I think this is just bad practice and also makes it difficult to support cross platform run configurations. I have the below `launchSettings.json` which works fine when I run it from Rider:

```json
{
  "profiles": {
    "My.Lambda.Function": {
      "commandName": "Executable",
      "commandLineArgs": "lambda-test-tool-8.0 --port 5050",
      "executablePath": "dotnet"
    }
  }
}
```

However, I'm not able to hit breakpoints when I launch this run configuration from Jetbrains Rider using Debug mode. Is there some way I can use the `dotnet` CLI tool to kick off the test tool but still be able to attach to it for debugging breakpoints?

Thanks in advance.

**Comment by ashishdhingra** [comment, marked as answer]
@rcdailey Good afternoon. Please verify if you followed instructions below:
- [Configure for JetBrains Rider](https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#configure-for-jetbrains-rider).
- We also updated section [Testing Executable Assemblies](https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#testing-executable-assemblies) to elaborate steps for debugging.

Thanks,
Ashish
